modules/ximgproc/src/structured_edge_detection.cpp

```diff
@@ -801,9 +801,11 @@ class StructuredEdgeDetectionImpl : public StructuredEdgeDetection
                 {// for j,k in [0;width)x[0;nTreesEval)
 
                     int currentNode = pIndex[j*nTreesEval + k];
-
-                    int start  = __rf.edgeBoundaries[currentNode];
-                    int finish = __rf.edgeBoundaries[currentNode + 1];
+                    size_t sizeBoundaries = __rf.edgeBoundaries.size();
+                    int convertedBoundaries = static_cast<int>(sizeBoundaries);
+                    int nBnds = (convertedBoundaries - 1) / (nTreesNodes * nTrees);
```

Comment:
Looks like `nBnds` is a constant for these loops (so it should be moved out).

Does "((convertedBoundaries - 1) % (nTreesNodes * nTrees)) == 0" always (remainder == 0)?
Or should we calculate original expression exactly (multiplication at first): `currentNode * (convertedBoundaries - 1) / (nTreesNodes * nTrees)`?